### ISSUE TITLE
Problem caused by GO_BUILD_CACHE

### ISSUE THREAD

**Issue by Andrewmatilde** [chaos-mesh/chaos-mesh#3686]
## Bug Report
This is a BUG in coding.
When we open builded(run make) chaos-mesh as a project in goland (In vscode ,this will also bring some strange problems to block download package auto).
Goland will run about 180+ times go list for file like go.mod.
So why we have only 4 go.mod in project but goland run 180+ times?
This is caused by fault in dev-env/Dockerfile.

```
        tmp_go_dir = f"{os.getenv('GO_BUILD_CACHE')}/chaos-mesh-gopath"
        tmp_go_build_dir = f"{os.getenv('GO_BUILD_CACHE')}/chaos-mesh-gobuild"

        pathlib.Path(tmp_go_dir).mkdir(parents=True, exist_ok=True)
        pathlib.Path(tmp_go_build_dir).mkdir(parents=True, exist_ok=True)
        cmd += ["--volume", f"{tmp_go_dir}:/tmp/go"]
        cmd += ["--volume", f"{tmp_go_build_dir}:/tmp/go-build"]
```

Here we map $(ROOT)/.cache/chaos-mesh/chaos-mesh-gopath as GOPATH in docker image , in docker image golang will download package in it.
But when we watch code outside the docker image, there will only add a .cache directory with a lot of go modules.
```
package main

import (
	"awesome0/.cache/chaos-mesh/chaos-mesh-gopath/pkg/mod/github.com/aws"
	"fmt"
)

func main() {
	fmt.Println(aws.A)
}
```
And .cache is the valid path golang modules , so all modules in .cache is valid parts of project chaos-mesh now.
I will try to fix it, because I am a goland user.

**Comment by STRRL** [comment]
I just realized that I had marked the `.cache` as `Excluded`:
<img width="769" alt="image" src="https://user-images.githubusercontent.com/20221408/194210863-59d4b9c3-5401-4d45-8019-7359794dca27.png">

But that configuration would saved in `.idea` folder, which ignored by git

**Comment by STRRL** [comment]
maybe we could create a docker volume to store the cache 

also cc @YangKeao

**Comment by Andrewmatilde** [comment]
> maybe we could create a docker volume to store the cache (go modules and go build cache)
> 
> also cc @YangKeao

But build with go mod tidy is already cached. I think docker build is usually used in cross building.

**Comment by YangKeao** [comment]
> I think docker build is usually used in cross building.

No. Docker build is used on any platform (even linux). I also developed the Chaos Mesh on Linux, and still use the docker build environment. It could solve a lot of problems: e.g. the customized goimport, the locked go version... The benefit of docker image is not to touch the global develop environment.

The replacement of docker image is something like bazel, nix-shell (flakes)...

> But build with go mod tidy is already cached.

It's not only used for local cache. It's also used for distribute cache in CI (e.g. e2e). Dumping all caches in a local directory helps to archive the cache and save them. (Well, this solution is also simular to bazel cache :cry: )

**Comment by Andrewmatilde** [comment]
> The benefit of docker image is not to touch the global develop environment.

But we cannot develop in global env now. I think this is the main problem. And I will add it.

> It's not only used for local cache. It's also used for distribute cache in CI (e.g. e2e). Dumping all caches in a local directory helps to archive the cache and save them. (Well, this solution is also simular to bazel cache 😢 )

I know it is for distribute cache in CI. But it is strange of not leaving a way for local cache.

**Comment by YangKeao** [comment]
> But we cannot develop in global env now. I think this is the main problem. And I will add it.

Good. I think it can be done with `IN_DOCKER=1` and some other options :thinking: .
